tests/testVariance.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 LSST/AURA
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""
+Tests for Variance measurement algorithm
+"""
+
+import unittest
+import lsst.utils.tests as utilsTests
+
+import lsst.pex.config as pexConfig
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.afw.image as afwImage
+import lsst.afw.detection as afwDetection
+import lsst.meas.base as measBase
+
+try:
+    display
+except NameError:
+    display = False
+
+
+class VarianceTest(unittest.TestCase):
+    def testVariance(self):
+        size = 128 # size of image (pixels)
+        center = afwGeom.Point2D(size//2, size//2) # object center
+        width = 1.0 # PSF width
+        flux = 10.0 # Flux of object
+        variance = 1.0 # Variance value
+
+        # Initial setup of an image
+        exp = afwImage.ExposureF(size, size)
+        image = exp.getMaskedImage().getImage()
+        mask = exp.getMaskedImage().getMask()
+        var = exp.getMaskedImage().getVariance()
+        image.set(0.0)
+        mask.set(0)
+        var.set(variance)
+
+        # Put down a PSF
+        psf = afwDetection.GaussianPsf(int(5*width), int(5*width), width)
+        exp.setPsf(psf)
+        psfImage = psf.computeImage(center).convertF()
+        psfImage *= flux
+        image.Factory(image, psfImage.getBBox(afwImage.PARENT)).__iadd__(psfImage)
+        var.Factory(var, psfImage.getBBox(afwImage.PARENT)).__iadd__(psfImage)
+
+        # Put in some bad pixels to ensure they're ignored
+        for i in range(-5, 6):
+            bad = size//2 + i*width
+            var.getArray()[bad, :] = float("nan")
+            mask.getArray()[bad, :] = mask.getPlaneBitMask("BAD")
+            var.getArray()[:, bad] = float("nan")
+            mask.getArray()[:, bad] = mask.getPlaneBitMask("BAD")
+
+        if display:
+            import lsst.afw.display.ds9 as ds9
```

Comment:
This looks like the old-school `afw.display` interface. I think the modern incantation would be more like:
```python
import lsst.afw.display as afwDisplay
afwDisplay.getDisplay().mtv(image, frame=1)
[etc]
```
See also [Doxygen](https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/afw_sec_display.html).